pilot/pkg/config/kube/gateway/conversion.go

```diff
@@ -114,32 +117,21 @@ func convertVirtualService(r *KubernetesResources, routeMap map[*k8s.HTTPRouteSp
 	}
 	for _, obj := range r.HTTPRoute {
 		route := obj.Spec.(*k8s.HTTPRouteSpec)
-		name := obj.Name + "-" + constants.KubernetesGatewayName
 
 		gateways, f := routeMap[route]
 		if !f {
+			// There are no gateways using this route
 			continue
 		}
-
-		httproutes := []*istio.HTTPRoute{}
```

Comment:
previous logic was wrong, and was merging hostnames across routes when hostname:route is 1:1 in the API